src/Data/Entity.php

```diff
@@ -13,25 +17,63 @@
 
     public function __construct($data = [])
     {
-        $this->data = $data;
+        $this->mapData($data);
     }
 
     public function getId()
     {
-        return $this->id;
+        return (int) $this->id;
+    }
+
+    public function has($field)
+    {
+        return array_key_exists($field, $this->properties) || in_array($field, $this->properties);
+    }
+
+    public function getDate($field)
+    {
+        return new Carbon($this->{$field});
+    }
+
+    public function getData()
+    {
+        $object = [];
+        foreach ($this->properties as $k=>$v) {
+            if (is_numeric($k)) {
+                $k = $v;
+                $v = null;
+            }
+            $object[$k] = $this->{$k};
+        }
+
+        return $object;
     }
 
     public function mapData(array $row)
     {
-        $this->data = $row;
+        foreach ($this->properties as $k=>$v) {
+            if (is_numeric($k)) {
+                $k = $v;
+                $v = null;
+            }
+            if (!array_key_exists($k, $row)) {
+                continue;
+            }
```

Comment:
I'll be honest... i don't understand what these 5 lines of code are supposed to be doing.